pkg/asset-manager-utils/test/SinglePoolAssetManager.test.ts

```diff
@@ -0,0 +1,211 @@
+  });
+
+  sharedBeforeEach('set up asset manager', async () => {
+    // Deploy tokens and vault
+    tokens = await TokenList.create(['DAI', 'MKR'], { sorted: true });
+    const vaultHelper = await Vault.create({ admin });
+    vault = vaultHelper.instance;
+
+    // deploy pool and add liquidity
+    const pool = await deploy('v2-vault/MockPool', { args: [vault.address, GeneralPool] });
+    poolId = await pool.getPoolId();
+
+    // set up asset manager
+    assetManager = await deploy('SinglePoolTestAssetManager', {
+      args: [vault.address, poolId, tokens.DAI.address],
+    });
+
+    await tokens.mint({ to: lp, amount: tokenInitialBalance });
+    await tokens.approve({ to: vault.address, from: [lp] });
+
+    // Assign assetManager to the DAI token, and other to the other token
+    const assetManagers = [assetManager.address, other.address];
+
+    await pool.registerTokens(tokens.addresses, assetManagers);
+
+    await vault.connect(lp).joinPool(poolId, lp.address, other.address, {
+      assets: tokens.addresses,
+      maxAmountsIn: tokens.addresses.map(() => MAX_UINT256),
+      fromInternalBalance: false,
+      userData: encodeJoin(
+        tokens.addresses.map(() => tokenInitialBalance),
+        tokens.addresses.map(() => 0)
+      ),
+    });
+  });
+
+  it('different managers can be set for different tokens', async () => {
+    expect((await vault.getPoolTokenInfo(poolId, tokens.DAI.address)).assetManager).to.equal(assetManager.address);
+    expect((await vault.getPoolTokenInfo(poolId, tokens.MKR.address)).assetManager).to.equal(other.address);
+  });
+
+  it('allows a pool controller to set the desired target investable %', async () => {
+    const targetPercentage = fp(0.2);
+    const poolController = lp; // TODO
+
+    const updatedConfig = { targetPercentage, criticalPercentage: 0, feePercentage: 0 };
+    await assetManager.connect(poolController).setPoolConfig(poolId, updatedConfig);
+
+    const result = await assetManager.getPoolConfig(poolId);
+    expect(result.targetPercentage).to.equal(updatedConfig.targetPercentage);
+    expect(result.criticalPercentage).to.equal(updatedConfig.criticalPercentage);
+    expect(result.feePercentage).to.equal(updatedConfig.feePercentage);
+  });
+
+  describe('when a token has been made investable', () => {
+    let poolController: SignerWithAddress; // TODO
+    const targetPercentage = fp(0.9);
+
+    beforeEach(async () => {
+      poolController = lp; // TODO
+      await assetManager
+        .connect(poolController)
+        .setPoolConfig(poolId, { targetPercentage, criticalPercentage: 0, feePercentage: 0 });
+    });
+
+    it('transfers only the requested token from the vault to the lending pool via the manager', async () => {
+      await expectBalanceChange(() => assetManager.connect(lp).capitalIn(poolId, amount), tokens, [
+        { account: assetManager.address, changes: { DAI: amount } },
+        { account: vault.address, changes: { DAI: -amount } },
```

Comment:
```suggestion
        { account: vault.address, changes: { DAI: amount.mul(-1) } },
```

Annoyingly, `-bn` breaks for large numbers (since they are converted to `Number`)

Reply:
(this is repeated in other places)